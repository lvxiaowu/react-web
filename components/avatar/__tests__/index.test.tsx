import React from 'react';
import { render, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Avatar from '../index';

describe('Avatar', () => {
  it('renders Avatars of different size correctly', () => {
    const wrapper = render(
      <div>
        <Avatar size="xl">XL</Avatar>
        <Avatar size="lg">LG</Avatar>
        <Avatar>MD</Avatar>
        <Avatar size="sm">SM</Avatar>
        <Avatar size="xs">XS</Avatar>
      </div>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Avatars of different shape correctly', () => {
    const wrapper = shallow(
      <div>
        <Avatar />
        <Avatar shape="circle" />
        <Avatar shape="square" />
      </div>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Avatars of different type correctly', () => {
    const wrapper = render(
      <div>
        <Avatar size="sm" style={{ fontSize: '12px' }}>
          USER
        </Avatar>
        <Avatar size="sm" style={{ color: 'red', backgroundColor: 'rgb(249, 232, 8)' }}>
          USER
        </Avatar>
        <Avatar
          size="sm"
          src="https://static-venus.shandiantech.com/skio/20211227/1640572420325_state.png"
          alt="my avatar"
        />
      </div>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Avatars correctly when set src, icon and string', () => {
    const wrapper = render(
      <div>
        <Avatar
          size="sm"
          src="https://static-venus.shandiantech.com/skio/20211227/1640572420325_state.png"
          alt="my avatar"
        />
      </div>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
