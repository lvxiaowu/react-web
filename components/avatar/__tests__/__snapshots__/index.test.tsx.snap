// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders Avatars correctly when set src, icon and string 1`] = `
<div>
  <span
    class="sd-avatar sd-avatar--circle sd-avatar--sm"
  >
    <img
      alt="my avatar"
      class="sd-avatar--image"
      src="https://static-venus.shandiantech.com/skio/20211227/1640572420325_state.png"
    />
  </span>
</div>
`;

exports[`Avatar renders Avatars of different shape correctly 1`] = `
<div>
  <Avatar
    prefixCls="sd-avatar"
    shape="circle"
    size="md"
  />
  <Avatar
    prefixCls="sd-avatar"
    shape="circle"
    size="md"
  />
  <Avatar
    prefixCls="sd-avatar"
    shape="square"
    size="md"
  />
</div>
`;

exports[`Avatar renders Avatars of different size correctly 1`] = `
<div>
  <span
    class="sd-avatar sd-avatar--circle sd-avatar--xl"
  >
    <span
      class="sd-avatar--string"
      style="transform:scale(1) translateX(-50%);-webkit-transform:scale(1) translateX(-50%);-ms-transform:scale(1) translateX(-50%)"
    >
      XL
    </span>
  </span>
  <span
    class="sd-avatar sd-avatar--circle sd-avatar--lg"
  >
    <span
      class="sd-avatar--string"
      style="transform:scale(1) translateX(-50%);-webkit-transform:scale(1) translateX(-50%);-ms-transform:scale(1) translateX(-50%)"
    >
      LG
    </span>
  </span>
  <span
    class="sd-avatar sd-avatar--circle sd-avatar--md"
  >
    <span
      class="sd-avatar--string"
      style="transform:scale(1) translateX(-50%);-webkit-transform:scale(1) translateX(-50%);-ms-transform:scale(1) translateX(-50%)"
    >
      MD
    </span>
  </span>
  <span
    class="sd-avatar sd-avatar--circle sd-avatar--sm"
  >
    <span
      class="sd-avatar--string"
      style="transform:scale(1) translateX(-50%);-webkit-transform:scale(1) translateX(-50%);-ms-transform:scale(1) translateX(-50%)"
    >
      SM
    </span>
  </span>
  <span
    class="sd-avatar sd-avatar--circle sd-avatar--xs"
  >
    <span
      class="sd-avatar--string"
      style="transform:scale(1) translateX(-50%);-webkit-transform:scale(1) translateX(-50%);-ms-transform:scale(1) translateX(-50%)"
    >
      XS
    </span>
  </span>
</div>
`;

exports[`Avatar renders Avatars of different type correctly 1`] = `
<div>
  <span
    class="sd-avatar sd-avatar--circle sd-avatar--sm"
    style="font-size:12px"
  >
    <span
      class="sd-avatar--string"
    >
      USER
    </span>
  </span>
  <span
    class="sd-avatar sd-avatar--circle sd-avatar--sm"
    style="color:red;background-color:rgb(249, 232, 8)"
  >
    <span
      class="sd-avatar--string"
      style="transform:scale(1) translateX(-50%);-webkit-transform:scale(1) translateX(-50%);-ms-transform:scale(1) translateX(-50%)"
    >
      USER
    </span>
  </span>
  <span
    class="sd-avatar sd-avatar--circle sd-avatar--sm"
  >
    <img
      alt="my avatar"
      class="sd-avatar--image"
      src="https://static-venus.shandiantech.com/skio/20211227/1640572420325_state.png"
    />
  </span>
</div>
`;
